tests/test_integration.py

```diff
@@ -293,49 +342,67 @@ def test_start_stop(self):
         self.check_vm_present()
 
     def test_disk_set(self):
-        def _get_disk_hv():
-            return (
-                self.vm.hypervisor.vm_sync_from_hypervisor(self.vm)
-                ['disk_size_gib']
-            )
-
-        def _get_disk_vm():
-            return parse_size(
-                self.vm.run("df -h / | tail -n+2 | awk '{ print $2 }'")
-                .strip(),
-                'G'
-            )
+        if self.datacenter_type == 'kvm.dct':
+            def _get_disk_hv():
+                return (
+                    self.vm.hypervisor.vm_sync_from_hypervisor(self.vm)
+                    ['disk_size_gib']
+                )
+
+            def _get_disk_vm():
+                return parse_size(
+                    self.vm.run("df -h / | tail -n+2 | awk '{ print $2 }'")
+                    .strip(),
+                    'G'
+                )
+        elif self.datacenter_type == 'aws.dct':
+            def _get_disk_vm():
+                partition = self.vm.run('findmnt -nro SOURCE /')
+                disk = self.vm.run('lsblk -nro PKNAME {}'.format(partition))
+                disk_size = self.vm.run(
+                    'lsblk -bdnro size /dev/{}'.format(disk))
+                disk_size_gib = int(disk_size) / 1024 / 1024 / 1024
+                return disk_size_gib
 
         # Initial size same as built
         obj = Query({'hostname': VM_HOSTNAME}, ['disk_size_gib']).get()
         size = obj['disk_size_gib']
-        self.assertEqual(_get_disk_hv(), size)
+        if self.datacenter_type == 'kvm.dct':
+            self.assertEqual(_get_disk_hv(), size)
         self.assertEqual(_get_disk_vm(), size)
 
         size = size + 1
         disk_set(VM_HOSTNAME, '+1')
         obj = Query({'hostname': VM_HOSTNAME}, ['disk_size_gib']).get()
 
         self.assertEqual(obj['disk_size_gib'], size)
-        self.assertEqual(_get_disk_hv(), size)
+        if self.datacenter_type == 'kvm.dct':
+            self.assertEqual(_get_disk_hv(), size)
         self.assertEqual(_get_disk_vm(), size)
 
-        size = 8
-        disk_set(VM_HOSTNAME, '{}GB'.format(size))
-        obj = Query({'hostname': VM_HOSTNAME}, ['disk_size_gib']).get()
-
-        self.assertEqual(obj['disk_size_gib'], size)
-        self.assertEqual(_get_disk_hv(), size)
-        self.assertEqual(_get_disk_vm(), size)
+        if self.datacenter_type == 'kvm.dct':
+            disk_set(VM_HOSTNAME, '{}GB'.format(size))
```

Comment:
is there no disk increasing for aws?!

Reply:
There is, but only one time, because then the disk is in optimizing state for unknown amount of time. I didn't want it to fail randomly.

Reply:
but is the unknown amount of time seconds or minutes or hours? i would find it better to test this still. if it really takes up to hours then we could still make the call and check for a valid response no?

Reply:
It is unknown for at least minutes, depending on the size of disk and increase. I'm checking the state of the volume now and if this is in state `optimizing` the disk_set raises an Exception, which I also check on the integration tests.